src/modules/campaign-monitor/services/campaign-monitor.service.ts

```diff
@@ -58,54 +66,173 @@ export class CampaignMonitorService {
 			}
 		});
 
-		if (recipients.length === 0) {
-			this.logger.error(
-				`Mail will not be sent - no recipients. emailInfo: ${JSON.stringify(emailInfo)}`
+		const data: CampaignMonitorData = {
+			to: recipients,
+			consentToTrack: 'unchanged',
+			data: this.convertVisitToEmailTemplateData(emailInfo.visit),
+		};
+
+		if (this.isEnabled) {
+			await this.sendTransactionalMail({
+				template: emailInfo.template,
+				data,
+			});
+		} else {
+			this.logger.log(
+				`Mock email sent. To: '${data.to}'. Template: ${
+					emailInfo?.template
+				}, data: ${JSON.stringify(data)}`
 			);
 			return false;
 		}
+		return true;
```

Comment:
this return true seems more logical to be placed after the sendTransactionalMail statement inside the if
so you have:
if 
  // send mail
  // return true => success
else 
  // handle error
  // return false => fail